litestar/testing/client/base.py

```diff
@@ -162,13 +162,16 @@ def _create_session_cookies(backend: ClientSideSessionBackend, data: dict[str, A
 
     async def _set_session_data(self, data: dict[str, Any]) -> None:
         mutable_headers = MutableScopeHeaders()
+        connection = fake_asgi_connection(
+            app=self.app,
+            cookies=dict(self.cookies),  # type: ignore[arg-type]
+        )
+        session_id = None
+        if self._session_backend is not None:
```

Comment:
You'll need to add a test that hit's this condition, i.e. one where `AsyncClient.set_session_data` / `SyncClient.set_session_data` is called without a session backend being set.

Reply:
Actually, could fix it by replacing it with `self.session_backend`: If the backend is None, an exception is raised, since you should not interatct with a backend that does not exist. Both mypy and tests accept that.